lib/python/picongpu/pypicongpu/laser.py

```diff
@@ -102,3 +108,80 @@ def _get_serialized(self) -> dict:
                 },
             },
         }
+
+
+@typeguard.typechecked
+class PlaneWaveLaser(Laser):
+    """
+    PIConGPU Plane Wave Laser
+
+    Holds Parameters to specify a plane wave laser
+    """
+
+    _name = "planewave"
+
+    class PolarizationType(enum.Enum):
+        """represents a polarization of a laser (for PIConGPU)"""
+
+        LINEAR = 1
+        CIRCULAR = 2
```

Comment:
Why are these class-level variables and not instance-level variables? 
Ffor these two variables, I get it, since they are global, but especially later (wavelength etc.) wouldn't make instance-level make more sense?

Reply:
@PrometheusPi I think here this is  just a python implementation of an enum class. And below these are actually member variables, pypicongpu uses currently pydantic models as base class. These are basically python dataclasess with type checks and this is the syntax for defining members.

Reply:
@pordyna is right, this is a class local definition of an enum

Reply:
but since we now support more than one laser the defintion should be moved outside of the laser definition.